cookbooks/aws-parallelcluster-slurm/resources/munge_key_manager.rb

```diff
@@ -23,35 +23,16 @@
 
 default_action :setup_munge_key
 
-def fetch_and_decode_munge_key(munge_key_secret_arn)
+def fetch_and_decode_munge_key
+  script_path = "#{node['cluster']['scripts_dir']}/slurm/update_munge_key.sh"
+
   declare_resource(:bash, 'fetch_and_decode_munge_key') do
     user 'root'
     group 'root'
-    cwd '/tmp'
-    code <<-FETCH_AND_DECODE
-      set -e
-      # Get encoded munge key from secrets manager
-      encoded_key=$(aws secretsmanager get-secret-value --secret-id #{munge_key_secret_arn} --query 'SecretString' --output text --region #{node['cluster']['region']})
-      # If encoded_key doesn't have a value, error and exit
-      if [ -z "$encoded_key" ]; then
-        echo "Error fetching munge key from Secrets Manager or the key is empty"
-        exit 1
-      fi
-
-      # Decode munge key and write to /etc/munge/munge.key
-      decoded_key=$(echo $encoded_key | base64 -d)
-      if [ $? -ne 0 ]; then
-        echo "Error decoding the munge key with base64"
-        exit 1
-      fi
-
-      echo "$decoded_key" > /etc/munge/munge.key
-
-      # Set ownership on the key
-      chown #{node['cluster']['munge']['user']}:#{node['cluster']['munge']['group']} /etc/munge/munge.key
-      # Enforce correct permission on the key
-      chmod 0600 /etc/munge/munge.key
-    FETCH_AND_DECODE
+    cwd File.dirname(script_path)
+    code <<-SHELL
+      ./#{File.basename(script_path)}
+    SHELL
```

Comment:
Please declare this as a resource of type `:execute`, and then use the same syntax as the `execute` resource.

Reply:
Done!